Snapyr/Classes/SnapyrSDK.m

```diff
@@ -41,11 +41,31 @@ + (void)setupWithConfiguration:(SnapyrSDKConfiguration *)configuration;
     });
 }
 
++ (void)handleNoticationExtensionRequestWithBestAttemptContent:(UNMutableNotificationContent * _Nonnull)bestAttemptContent originalRequest:(UNNotificationRequest *_Nonnull)originalRequest contentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler
+{
+    [SnapyrSDK handleNoticationExtensionRequestWithBestAttemptContent:bestAttemptContent originalRequest:originalRequest contentHandler:contentHandler devMode:NO];
+}
+
 + (void)handleNoticationExtensionRequestWithWriteKey:(NSString *)writeKey bestAttemptContent:(UNMutableNotificationContent * _Nonnull)bestAttemptContent originalRequest:(UNNotificationRequest *_Nonnull)originalRequest contentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler
 {
     [SnapyrSDK handleNoticationExtensionRequestWithWriteKey:writeKey bestAttemptContent:bestAttemptContent originalRequest:originalRequest contentHandler:contentHandler devMode:NO];
 }
 
++ (void)handleNoticationExtensionRequestWithBestAttemptContent:(UNMutableNotificationContent * _Nonnull)bestAttemptContent originalRequest:(UNNotificationRequest *_Nonnull)originalRequest contentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler devMode:(BOOL)enableDevMode
+{
+    NSString *writeKey = __sharedInstance.configuration.writeKey;
+    if (!writeKey) {
+        writeKey = [SnapyrUtils getWriteKey];
+    }
```

Comment:
Note that this is a static method that's meant to be called from a notification service extension, which is an isolated process from the main app. That means that in the context of this method, there was not a previous SDK instance initiated - the write key will always need to be fetched from storage.

We made some updates to preference storage in order to support the notification service extension stuff, by requiring the SDK implementor to set up an "app group" for both the main app and the extension, and then use shared sandbox storage so that both sides can access the same data.

You can use the utility function `getGroupUserDefaults()` to get the shared defaults without going through boilerplate for that again.

See some related commits:
https://github.com/snapyrautomation/snapyr-ios-sdk/commit/7311acf774de13bbbbc08cc1e2f5e2b7008ba9ee
https://github.com/snapyrautomation/snapyr-ios-sdk/commit/b6f1290460b5a5e4a73c38145a472fb04310e90f
https://github.com/snapyrautomation/snapyr-ios-sdk/commit/cb7ff457a5ec428b218a90b82d0752ccabe993d6